jobs/kpi-forecasting/README.md

```diff
@@ -1,66 +1,85 @@
 
-## Usage
+# Usage
 
-This script is intended to be run in a docker container.
-Build the docker image with:
+### Docker Container
+
+This job is intended to be run in a Docker container. If you're not familiar with Docker, it can be helpful to first install
+[Docker Desktop](https://docs.docker.com/desktop/) which provides a GUI.
+
+First, ensure that you have `CLOUDSDK_CONFIG` set as a environment variable for your shell so that Docker can find your gcloud credentials.
+The default is `~/.config/gcloud`:
 
 ```sh
-docker build -t kpi-forecasting .
+export CLOUDSDK_CONFIG="~/.config/gcloud"
 ```
 
-To run locally, install dependencies with:
+To [build or re-build the Docker image](https://docs.docker.com/engine/reference/commandline/compose_build/), run the following command from the top-level `kpi-forecasting`. To force Docker to rebuild from scratch, pass the `--no-cache` flag.
 
 ```sh
-pip install -r requirements.txt
+docker compose build
 ```
 
-Run the scripts with:
+Start a container from the Docker image with the following command:
 
 ```sh
-python ~/kpi-forecasting/kpi_forecasting.py -c ~/kpi-forecasting/yaml/desktop_non_cumulative.yaml
+docker compose up
+```
 
-python ~/kpi-forecasting/kpi_forecasting.py -c ~/kpi-forecasting/yaml/mobile_non_cumulative.yaml
+A metric can be forecasted by using a command line argument that passes the relevant YAML file to the `kpi_forecasting.py` script.
+[Here are approaches for accessing a Docker container's terminal](https://docs.docker.com/desktop/use-desktop/container/#integrated-terminal).
+
+For example, the following command forecasts Desktop DAU numbers:
+
+```sh
+python ~/kpi-forecasting/kpi_forecasting.py -c ~/kpi-forecasting/configs/dau_desktop.yaml
 ```
 
-### On SQL Queries And Preprocessing
+### Local Python
 
-One of the challenges in automating multiple forecasts is that data formats may not always be consistent. This is a challenge precisely because Prophet demands that data conform to a very strict standard of two columns; 'ds' and 'y' as its primary inputs. Other columns can be added as regressors, but often these are trivial to implement in Python but challenging to implement in SQL, or vice versa, depending on their specifics.
+You can also run the code outside of a Docker container. The code below creates a new Conda environment called `kpi-forecasting-dev`.
+It assumes you have Conda installed. If you'd like to run the code in a Jupyter notebook, it is handy to install Jupyter in your `base` environment.
+The `ipykernel` commands below will ensure that the `kpi-forecasting-dev` environment is made available to Jupyter.
 
-Rather than try and force things to conform to an ultra-strict standard to solve a wide variety of forecasts, this repo focuses ONLY on KPIs, and is organized in such a way to make that easier to maintain.
+```sh
+conda create --name kpi-forecasting-dev python=3.10 pip ipykernel
```

Comment:
Is the virtual environment created in the project directory by conda? If yes, perhaps we should include it in the `.gitignore`

Reply:
Done. By default conda adds environments to a common directory (`~/Users/username/conda/envs`) but it's easy enough to add `envs` to `.gitignore` in case we choose to store environments in the repo in the future.